packages/charts/src/components/legend/legend.tsx

```diff
@@ -0,0 +1,203 @@
+import { isArray, isString } from 'lodash'
+import {
+  type ComponentPropsWithoutRef,
+  forwardRef,
+  useCallback,
+  useImperativeHandle,
+  useMemo,
+  useState,
+} from 'react'
+import { defaultColorPreset } from '../../theme/colors'
+import type { EChartsOption } from 'echarts'
+import '../../theme/components/legend.css'
+import {
+  changeBarRoundingExclusive,
+  changeBarRoundingToogle,
+  checkAllSelected,
+} from '../../utils/legend'
+import type ReactECharts from 'echarts-for-react'
+import { LegendItem } from './legend-item'
+import type { ChartConfig } from '../../types/chart'
+
+export const Legend = forwardRef<LegendHandle, LegendProps>(
+  function Legend(props, ref) {
```

Comment:
It's little bit confuse te name of Legend is used twice. Could you change to arrow function instead of?